coregrind/m_gdbserver/riscv64-cpu-valgrind-s1.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0"?>
+<!-- Copyright (C) 2018-2022 Free Software Foundation, Inc.
+
+     Copying and distribution of this file, with or without modification,
+     are permitted in any medium without royalty provided the copyright
+     notice and this notice are preserved.  -->
+
+<!-- Register numbers are hard-coded in order to maintain backward
+     compatibility with older versions of tools that didn't use xml
+     register descriptions.  -->
```

Comment:
The comment is misleading because it should not be needed to hardcode register numbers on the Valgrind side. Commit [172fb711a23](https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=172fb711a2336b62cf8d58bbb4b27dc71ca8c02d) ("gdb/riscv: Use legacy register numbers in default target description") in binutils-gdb explains that it was necessary on the GDB client to correctly support targets which used the previous hardcoded numbers.